activemodel/lib/active_model/secure_password.rb

```diff
@@ -97,7 +121,20 @@ def has_secure_password(attribute = :password, validations: true)
         #   user.authenticate_password('mUc3m00RsqyRe') # => user
         define_method("authenticate_#{attribute}") do |unencrypted_password|
           attribute_digest = send("#{attribute}_digest")
-          BCrypt::Password.new(attribute_digest).is_password?(unencrypted_password) && self
+          password = BCrypt::Password.new(attribute_digest)
+          if password.is_password?(unencrypted_password)
+            if password.cost < ActiveModel::SecurePassword.cost
+              rehashed_password = BCrypt::Password.create(unencrypted_password, cost: ActiveModel::SecurePassword.cost)
+              if respond_to? :update_column
+                update_column "#{attribute}_digest", rehashed_password
+              else
+                send "#{attribute}_digest=", rehashed_password
+              end
```

Comment:
Active Model doesn't know anything about how to tell a model to persist itself. Glaring mismatch here 😣

May need to move some of this responsibility to AR instead. But then it's confusing for other AM users when the feature doesn't tap into their model's lifecycle.

Reply:
After https://github.com/codahale/bcrypt-ruby/pull/181 I had a PR for rehashing in the works but was nowhere near done. This is one of the questions I had as well and I figured this is the better way to go about it: having a bunch of callbacks trigger just because someone authenticated ("you changed your password" emails) seemed like quite a surprising side effect for most folks. 

In my WIP PR I was planning to have a `updated_at` be updated as well, but that might be Shopify practice carrying over.